core/commonMain/src/kotlinx/serialization/internal/Primitives.kt

```diff
@@ -74,170 +74,58 @@ private fun checkName(serialName: String) {
 internal fun <T : Any> KClass<T>.builtinSerializerOrNull(): KSerializer<T>? =
     BUILTIN_SERIALIZERS[this] as KSerializer<T>?
 
-@Deprecated(
-    message = "Deprecated in the favour of Unit.serializer() extension",
-    level = DeprecationLevel.ERROR,
-    replaceWith = ReplaceWith("Unit.serializer()", imports = ["kotlinx.serialization.builtins.serializer"])
-)
-@Suppress("DEPRECATION_ERROR")
-public object UnitSerializer : KSerializer<Unit> by ObjectSerializer("kotlin.Unit", Unit)
+internal object UnitSerializer : KSerializer<Unit> by ObjectSerializer("kotlin.Unit", Unit)
 
-@Deprecated(
-    message = "Deprecated in the favour of Boolean.serializer() extension",
-    level = DeprecationLevel.ERROR,
-    replaceWith = ReplaceWith("Boolean.serializer()", imports = ["kotlinx.serialization.builtins.serializer"])
-)
-public object BooleanSerializer : KSerializer<Boolean> {
+internal object BooleanSerializer : KSerializer<Boolean> {
```

Comment:
For the additional reassurance, lets mark them with `@PublishedApi` too — they can be used in plugin-generated code

Reply:
Sure